### PR TITLE
Wait to dispose RequestAborted CTS

### PR DESCRIPTION
**Pull request by halter73**
Partially Addresses #4422

### CODE REVIEW

**Comment by halter73** [comment]
@muratg I would like to get this into a 2.2 patch.

**Review by natemcmaster — CHANGES_REQUESTED** [review]
Can you make sure the affected assemblies are added to the list in https://github.com/aspnet/AspNetCore/blob/release/2.2/eng/PatchConfig.props?

File: src/Servers/Kestrel/Core/test/Http1ConnectionTests.cs [hunk 1]
```diff
@@ -724,6 +728,22 @@ public async Task RequestAbortedTokenIsResetBeforeLastWriteWithChunkedEncoding()
...
+            var originalRegistration = originalToken.Register(() => { });
+
+            _http1Connection.Abort(new ConnectionAbortedException());
+
+            Assert.True(originalToken.WaitHandle.WaitOne(TestConstants.DefaultTimeout));
+            Assert.True(_http1Connection.RequestAborted.WaitHandle.WaitOne(TestConstants.DefaultTimeout));
+
+#if NETCOREAPP2_2
+            Assert.Equal(originalToken, originalRegistration.Token);
+#endif
```
Comment (Tratcher) [inline src/Servers/Kestrel/Core/test/Http1ConnectionTests.cs]: #else TFMs need updating.

**Review by Tratcher — APPROVED** [review]

**Comment by halter73** [comment]
@natemcmaster It's nice not to have to switch repos anymore.

**Review by natemcmaster — APPROVED** [review]

**Comment by vivmishra** [comment]
Approved for 2.2.2. 

We will take the CoreCLR fix still in 2.2.1. 
https://github.com/dotnet/coreclr/pull/21417

File: eng/PatchConfig.props [hunk 2]
```diff
@@ -20,6 +20,7 @@ Later on, this will be checked using this condition:
       Microsoft.AspNetCore.Server.IIS;
       Microsoft.AspNetCore.Server.IISIntegration;
       Microsoft.AspNetCore.Server.IntegrationTesting.IIS;
+      Microsoft.AspNetCore.Server.Kestrel.Core;
```
Comment (dougbu) [inline eng/PatchConfig.props]: No, your `release/2.2` branch is behind. Please rebase then add this to the new block that's for 2.2.2.

**Pull request merged**